### PR TITLE
Make "consuming messages with a custom assignment strategy" stable

### PR DESCRIPTION
**Pull request by abicky**
In the example, one consumer sometimes consumers all messages before another consumer joins. As a result, the example is unstable. This PR makes it stable by ensuring all consumers have joined.

### CODE REVIEW

File: spec/functional/consumer_group_spec.rb [hunk 1]
```diff
@@ -476,12 +476,21 @@ def call(cluster:, members:, partitions:)
       end
     end
 
+    joinined_consumers = []
```
Comment (dasch) [inline spec/functional/consumer_group_spec.rb:479]:
```suggestion
    joined_consumers = []
```

File: spec/functional/consumer_group_spec.rb [hunk 2]
```diff
@@ -476,12 +476,21 @@ def call(cluster:, members:, partitions:)
...
     consumers = 2.times.map do |i|
       assignment_strategy = assignment_strategy_class.new(i + 1)
 
       kafka = Kafka.new(kafka_brokers, client_id: "test", logger: logger)
       consumer = kafka.consumer(group_id: group_id, offset_retention_time: offset_retention_time, assignment_strategy: assignment_strategy)
       consumer.subscribe(topic)
+
+      allow(consumer).to receive(:trigger_heartbeat).and_wrap_original do |m, *args|
+        joinined_consumers |= [consumer]
```
Comment (dasch) [inline spec/functional/consumer_group_spec.rb:488]:
```suggestion
        joined_consumers |= [consumer]
```

File: spec/functional/consumer_group_spec.rb [hunk 3]
```diff
@@ -476,12 +476,21 @@ def call(cluster:, members:, partitions:)
...
+        # Wait until all the consumers try to join to prevent one consumer from processing all messages
+        raise Kafka::HeartbeatError if joinined_consumers.size < consumers.size
```
Comment (dasch) [inline spec/functional/consumer_group_spec.rb:490]:
```suggestion
        raise Kafka::HeartbeatError if joined_consumers.size < consumers.size
```

**Review by dasch — CHANGES_REQUESTED** [review]
Small typo, but otherwise :+1:

**Comment by abicky** [comment]
Thank you for your review! I applied your suggestions in https://github.com/zendesk/ruby-kafka/pull/876/commits/fe7bc77212bfdff71cc3705ea68900e328922f9d.

**Pull request merged**